File: native_security/ledger_keyring/android/src/main/kotlin/com/chapter2/ledger/models/LedgerClearSignPrompt.kt
package com.chapter2.ledger.models

public data class ClearSignField(
    val label: String,
    val value: String,
    val isCritical: Boolean = false
)

public enum class RiskTier(val displayName: String) {
    BENIGN("BENIGN"),
    ELEVATED("ELEVATED"),
    HIGH_RISK("HIGH_RISK"),
    CRITICAL("CRITICAL");

    companion object {
        fun fromScore(score: UByte): RiskTier = when (score.toUInt()) {
            in 0u..<50u -> BENIGN
            in 50u..<75u -> ELEVATED
            in 75u..<90u -> HIGH_RISK
            else -> CRITICAL
        }
    }
}

public data class LedgerClearSignPrompt(
    val title: String = "CHAPTER 2 TREASURY ESCALATION",
    val actionId: String,
    val fields: List<ClearSignField>,
    val digestHex: String,
    val riskTier: RiskTier,
    val timestamp: Long = System.currentTimeMillis()
)

File: native_security/ledger_keyring/ios/Sources/LedgerKeyring/Models/LedgerClearSignPrompt.swift
import Foundation

public struct ClearSignField: Equatable, Codable, Sendable {
    public let label: String
    public let value: String
    public let isCritical: Bool

    public init(label: String, value: String, isCritical: Bool = false) {
        self.label = label
        self.value = value
        self.isCritical = isCritical
    }
}

public enum RiskTier: String, Codable, Sendable {
    case benign = "BENIGN"
    case elevated = "ELEVATED"
    case highRisk = "HIGH_RISK"
    case critical = "CRITICAL"

    public static func from(score: UInt8) -> RiskTier {
        switch score {
        case 0..<50:
            return .benign
        case 50..<75:
            return .elevated
        case 75..<90:
            return .highRisk
        default:
            return .critical
        }
    }
}

public struct LedgerClearSignPrompt: Equatable, Codable, Sendable {
    public let title: String
    public let actionId: String
    public let fields: [ClearSignField]
    public let digestHex: String
    public let riskTier: RiskTier
    public let timestamp: Date

    public init(
        title: String = "CHAPTER 2 TREASURY ESCALATION",
        actionId: String,
        fields: [ClearSignField],
        digestHex: String,
        riskTier: RiskTier,
        timestamp: Date = Date()
    ) {
        self.title = title
        self.actionId = actionId
        self.fields = fields
        self.digestHex = digestHex
        self.riskTier = riskTier
        self.timestamp = timestamp
    }
}
